src/main/java/org/jabref/logic/importer/fetcher/DoiResolution.java

```diff
@@ -70,6 +72,11 @@
                         LOGGER.info("Fulltext PDF found @ " + sciLink);
                         pdfLink = links.get(0);
                     }
+                } catch (UnsupportedMimeTypeException type) {
+                    // this might be the PDF already as we follow redirects
+                    if (type.getMimeType().startsWith("application/pdf")) {
+                        return Optional.of(new URL(type.getUrl()));
+                    }
```

Comment:
else log warning / rethrow?

Reply:
Is this information relevant for us? We did not get a URL then, so we just return an empty optional (as before).